crates/sparrow-runtime/src/prepare.rs

```diff
@@ -46,30 +38,90 @@ const UPLOAD_BUFFER_SIZE_IN_BYTES: usize = 5_000_000;
 /// Returns a fallible iterator over pairs containing the data and metadata
 /// batches.
 pub async fn prepared_batches<'a>(
+    object_stores: &ObjectStoreRegistry,
     source_data: &SourceData,
     config: &'a TableConfig,
     slice: &'a Option<slice_plan::Slice>,
 ) -> error_stack::Result<BoxStream<'a, error_stack::Result<(RecordBatch, RecordBatch), Error>>, Error>
 {
-    let prepare_hash = get_prepare_hash(source_data)?;
+    // let prepare_hash = get_prepare_hash(source_data)?;
     let prepare_iter = match source_data.source.as_ref() {
         None => error_stack::bail!(Error::MissingField("source")),
         Some(source) => match source {
             source_data::Source::ParquetPath(source) => {
-                let reader = open_file(source)?;
-                let file_size = reader
-                    .metadata()
-                    .into_report()
-                    .change_context(Error::Internal)?
-                    .len();
-                reader_from_parquet(config, reader, file_size, prepare_hash, slice).await?
+                let url = ObjectStoreUrl::from_str(source)
+                    .change_context(Error::InvalidUrl(source.to_owned()))?;
+                let file = ParquetFile::try_new(object_stores, url, None)
+                    .await
+                    .change_context(Error::CreateReader)?;
+
+                let file_size = file.object_meta.size;
+                let num_rows = file.num_rows();
+                let num_files = get_num_files(file_size);
```

Comment:
this'll be the number of prepared files we produce from the source file, correct?

Reply:
This is the number of files we would like to prepare, yes.